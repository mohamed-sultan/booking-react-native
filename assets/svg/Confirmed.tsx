import * as React from 'react';
import Svg, { SvgProps, Rect, Path } from 'react-native-svg';
import { memo } from 'react';

const SvgComponent = (props: SvgProps) => {
  const { width = 90, height = 89, ...rest } = props;

  return (
    <Svg width={width} height={height} fill="none" {...rest}>
      <Rect width={90} height={89} fill="#03B57C" rx={44.5} />
      <Path
        fill="#fff"
        d="M86.758 41.217c1.623 1.498 1.623 4.118 0 5.616l-31.95 31.95c-1.498 1.623-4.118 1.623-5.616 0L33.217 62.808c-1.623-1.498-1.623-4.118 0-5.616 1.498-1.623 4.118-1.623 5.616 0l13.23 13.104 29.079-29.08c1.498-1.622 4.118-1.622 5.616 0Z"
        scale={0.75} // scales the path proportionally (optional)
      />
    </Svg>
  );
};

const Confirmed = memo(SvgComponent);
export default Confirmed;
