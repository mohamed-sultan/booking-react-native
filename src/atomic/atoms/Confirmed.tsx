import React from 'react';
import {View} from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import {lightColors} from '@constants/colors';

const Confirmed = ({size}: any) => {
  return (
    <View>
      <Icon name="checkmark-circle" size={size} color={lightColors.success} />
    </View>
  );
};

export default React.memo(Confirmed);
