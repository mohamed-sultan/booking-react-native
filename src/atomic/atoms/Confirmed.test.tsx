import React from 'react';
import { render } from '@testing-library/react-native';
import Confirmed from './Confirmed';

// Mock react-native-vector-icons/Ionicons to avoid rendering issues
jest.mock('react-native-vector-icons/Ionicons', () => {
  return (props: any) => {
    // Return a simple View with testID for snapshot
    return <mock-icon {...props} />;
  };
});

// Mock the colors module
jest.mock('@constants/colors', () => ({
  lightColors: {
    success: '#00FF00',
  },
}));

describe('Confirmed', () => {
  it('renders correctly with given size', () => {
    const { toJSON } = render(<Confirmed size={32} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
