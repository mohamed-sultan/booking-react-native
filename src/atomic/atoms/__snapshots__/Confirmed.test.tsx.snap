// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmed renders correctly with given size 1`] = `
<View>
  <mock-icon
    color="#00FF00"
    name="checkmark-circle"
    size={32}
  />
</View>
`;
